ee/tabby-ui/app/search/components/search.tsx

```diff
@@ -227,27 +231,48 @@ export function Search() {
   const repositoryList = data?.repositoryList
 
   const [afterCursor, setAfterCursor] = useState<string | undefined>()
-  const [{ data: threadMessages, fetching: fetchingMessages }] = useQuery({
+  const [
+    {
+      data: threadMessages,
+      error: threadMessagesError,
+      fetching: fetchingMessages
+    }
+  ] = useQuery({
     query: listThreadMessages,
     variables: {
       threadId: threadId as string,
-      first: 20,
+      first: PAGE_SIZE,
       after: afterCursor
     },
     pause: !threadId || isReady
   })
 
-  // todo scroll and setAfterCursor
   useEffect(() => {
     if (threadMessages?.threadMessages?.edges?.length) {
       const messages = threadMessages.threadMessages.edges
         .map(o => o.node)
         .slice()
-      setMessages(messages)
-      setIsReady(true)
+      setMessages(prev => [...prev, ...messages])
+    }
+
+    if (threadMessages?.threadMessages) {
+      const hasNextPage = threadMessages?.threadMessages?.pageInfo?.hasNextPage
+      const endCursor = threadMessages?.threadMessages.pageInfo.endCursor
+      if (hasNextPage && endCursor) {
```

Comment:
After fetching all messages, the message list and prompt form are displayed. Before this, a skeleton is shown